File: app/channels/chat_channel.rb
class ChatChannel < ApplicationCable::Channel

  def subscribed
    stream_from "chatclient"
  end

  def receive(data)
    # Parse the JSON data and extract the message content
    message = JSON.parse(data)["message"]

    # Send the message to the appropriate recipient using Action Cable's broadcast method
    broadcast_to("chatclient", { message: message })
  end
  end

  def unsubscribed
    # Any cleanup needed when channel is unsubscribed
  end

File: app/javascript/channels/chat_channel.js
import consumer from "channels/consumer"

consumer.subscriptions.create("ChatChannel", {
  connected() {
    // Called when the subscription is ready for use on the server
  },

  disconnected() {
    // Called when the subscription has been terminated by the server
  },

  received(data) {
    // Called when there's incoming data on the websocket for this channel
    // Parse the JSON data and extract the message content
    const message = JSON.parse(data)["message"];

    // Append the message to the chat window
    const chatContainer = document.getElementById("chat-container");
    const newMessageElement = document.createElement("div");
    newMessageElement.textContent = message;
    chatContainer.appendChild(newMessageElement);
  }
});
